triton/resource_snapshot_test.go

```
@@ -0,0 +1,112 @@
+package triton
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/joyent/triton-go/compute"
+)
+
+func TestAccTritonSnapshot_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckTritonSnapshotDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTritonSnapshotConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckTritonSnapshotExists("triton_snapshot.test"),
+					func(*terraform.State) error {
+						time.Sleep(30 * time.Second)
+						return nil
+					},
+				),
+			},
+		},
+	})
+}
+
+func testCheckTritonSnapshotExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+		conn := testAccProvider.Meta().(*Client)
+		c, err := conn.Compute()
+		if err != nil {
+			return err
+		}
+
+		snapshot, err := c.Snapshots().Get(context.Background(), &compute.GetSnapshotInput{
+			Name:      rs.Primary.ID,
+			MachineID: rs.Primary.Attributes["machine_id"],
+		})
+		if err != nil {
+			return fmt.Errorf("Bad: Check Snapshot Exists: %s", err)
+		}
+
+		if snapshot == nil {
+			return fmt.Errorf("Bad: Snapshot %q does not exist", rs.Primary.ID)
```

This is a Terraform testing question but why does this return an error instead of returning `false` or `nil` for the check? Just curious.

---

let's talk about this offline - this is something I have feelings about ;)

---

Fair enough.